### DISCUSSION TITLE
iOS simulator mock location methods

### DISCUSSION THREAD

**Discussion by adelZilberman0** [appium/appium#19293] [Q&A] [answered]
Hi everyone, I'm building an Appium automation project for my app.
In order to test the app I need to set the location to different places in each test I execute. (on IOS simulator)
I have tried everything I found on the internet and nothing seems to work:
1. driver.setLocation() doesn't seem to be supported by appium. getting a syntax error as if it doesn't exist...
2. executeScript doesn't seem to work either. 
driver.executeScript("mobile: setLocation",
                "{ \"latitude\": " + latitude + ", \"longitude\": " + longitude + " }"); 
getting this error -> org.openqa.selenium.UnsupportedCommandException

The only option left is to mock location through the simulator itself but it's not practical at all since I need to change it a few times while the automation runs...

Does anyone know of a way I could make this work? Or did anyone encounter such a problem?

**Comment by jlipps** [comment, marked as answer]
I believe I replied to this in another location, but the second parameter to `executeScript` should not be a string but rather a map. The Java client does the conversion to JSON for you, so you must provide Java objects to the method, not JSON strings.

**Comment by adelZilberman0** [comment]
@jlipps thank you!
